gpt_bpe_test.go

```diff
@@ -787,25 +790,102 @@ func TestLlamaEncoder_Encode(t *testing.T) {
+	output := llama2Encoder.Decode(&llamaTokens)
 	assert.Equal(t, outputString, output)
 }
 
 func TestLlamaTwoEncodeDecode(t *testing.T) {
 	testString := "The fox jumped over the hare.\nThe turtle is faster than the hare."
 	outputString := "The fox jumped over the hare.\nThe turtle is faster than the hare."
-	llamaTokens := Llama2Encoder.Encode(&testString)
-	output := Llama2Encoder.Decode(llamaTokens)
+	llamaTokens := llama2Encoder.Encode(&testString)
+	output := llama2Encoder.Decode(llamaTokens)
 	assert.Equal(t, outputString, output)
 }
 
+func TestMistralEncoder_Encode(t *testing.T) {
+	testString := "The fox jumped over the hare.\nThe turtle is faster than the hare."
+	mistralTokens := mistralEncoder.Encode(&testString)
+	fmt.Printf("mistralTokens: %v\n", mistralTokens)
+	assert.Equal(t, mistralTokens, &Tokens{1, 415, 285, 1142, 14949, 754, 272, 295, 492, 28723, 13, 1014, 261, 3525, 291, 349, 9556, 821, 272, 295, 492, 28723})
+}
+
+func TestMistralTokenizerDecode(t *testing.T) {
+	outputString := "<s> The fox jumped over the hare.\nThe turtle is faster than the hare."
+	mistralTokens := Tokens{1, 415, 285, 1142, 14949, 754, 272, 295, 492, 28723, 13, 1014, 261, 3525, 291, 349, 9556, 821, 272, 295, 492, 28723}
+	output := mistralEncoder.Decode(&mistralTokens)
+	assert.Equal(t, outputString, output)
+}
+
+func TestMistralEncodeDecode(t *testing.T) {
+	testString := "The fox jumped over the hare.\nThe turtle is faster than the hare."
+	outputString := "<s> The fox jumped over the hare.\nThe turtle is faster than the hare."
+	mistralTokens := mistralEncoder.Encode(&testString)
+	output := mistralEncoder.Decode(mistralTokens)
+	assert.Equal(t, outputString, output)
+}
+
+func TestMistralEncodeDecodeFrankenstein(t *testing.T) {
+	frankensteinCorpus := "resources/frankenstein.txt"
+	frankensteinText, err := os.ReadFile(frankensteinCorpus)
+	if err != nil {
+		t.Errorf("Error reading Frankenstein corpus: %v", err)
+	}
+	frankensteinString := string(frankensteinText)
+	mistralTokens := mistralEncoder.Encode(&frankensteinString)
+	output := mistralEncoder.Decode(mistralTokens)
+	assert.Equal(t, "<s> "+frankensteinString, output)
+}
+
+func TestReadTokenizerConfig(t *testing.T) {
+	fmt.Println("Testing ReadTokenizerConfig")
+	// json with eos, bos, pad as strings
+	jsonStr := `{"eos_token": "TC", "bos_token": "TD", "pad_token": "TE"}` //cooresponds to 6669, 10989, 5428 in pythia vocab
+
+	//download filler model
+	modelId := "EleutherAI/pythia-70m"
+	destPath := "./TestReadTokenizerConfig"
+	destPathPTR := &destPath
+	defer os.RemoveAll(destPath)
+	var rsrcType resources.ResourceType
+	rsrcType = resources.RESOURCETYPE_TRANSFORMERS
+	hfApiToken := os.Getenv("HF_API_TOKEN")
+	os.MkdirAll(destPath, 0755)
+	_, rsrcErr := resources.ResolveResources(modelId, destPathPTR,
+		resources.RESOURCE_MODEL, rsrcType, hfApiToken)
+	if rsrcErr != nil {
+		os.RemoveAll(destPath)
+		t.Errorf("Error downloading model resources: %s", rsrcErr)
+	}
```

Comment:
Are you sure that extra `os.RemoveAll(destPath)` is necessary when you're deferring it in the outer scope? Is there any risk of `t.Errorf` not returning or something?

```suggestion
	if rsrcErr != nil {
		t.Errorf("Error downloading model resources: %s", rsrcErr)
	}
```

Reply:
You're right, if it's deferred, we won't need to call it again.